### ISSUE TITLE
Difference between number of activities between students

### ISSUE THREAD

**Issue by lapega** [deraadt/moodle-block_completion_progress#44]
In the general view, the progress bar show different number of activities between students.

![Captura de pantalla de 2019-08-06 17-30-31](https://user-images.githubusercontent.com/572502/62575067-4aa9ca00-b870-11e9-97bc-eefc00f03e9a.png)

**Comment by deraadt** [comment]
Yes, that is correct. The teacher sees what the student sees and that can vary between students depending on what is visible to them.

**Comment by lapega** [comment]
Hi @deraadt , thanks for the rapid response!
I think there's a wrong behavior. I'll explain:

I've configured the plugin to show only some kind of activities. These activities are only available if the student finished the previous one. In my case, the have Act. 1, 2, ... 7, they only can access to activity 3 if they finished activity 2.
On the other hand I've set the course in 4 sections (weeks), so the students only can view the activities in the previous and active week.

If you look at the image previously attached, the students with **82%** have less blocks that the others. When mouse-hovering the progress I can see the missing activity, and it is the number **5**. The students with **82%** are the ones that finished the activity **4** and now they have access to activity **5**, but the other students with less of 82% haven't finished activity **4** so they are not allowed to access activity **4**, and here is where I think there's something wrong with the plugin.
A thing of interest, is that the activity **5** is in the next week module, so no student can't access that module until the next week, even if they are in the **82%** group.

**Summarizing**: the students who have access to a module in a hidden section won't see that module in the progress bar, but students who aren't allowed to that module, can see that block in the progress bar.
